probe-rs/src/probe/mod.rs

```diff
@@ -45,13 +45,13 @@ pub enum DebugProbeError {
     // TODO: This is core specific, so should probably be moved there.
     #[error("Operation timed out.")]
     Timeout,
-    #[error("Communication with access port failed: {0:?}")]
-    AccessPort(#[from] AccessPortError),
+    #[error("An error specific to the selected architecture occured: {0:?}")]
```

Comment:
I think we should use the non-debug print here, this is for the `Display` implementation after all.
```suggestion
    #[error("An error specific to the selected architecture occured: {0}")]
```